docs/source/architecture/rest_api.rst

```diff
@@ -2,313 +2,47 @@
 
+The JSON response envelope also includes the details needed to parse errors. For
+more information, see the :doc:`../rest_api/error_codes`.
 
 Pagination
-----------
-All endpoints that return lists of resources will automatically paginate
-results, limited by default to 1000 resources. In order to specify what range of
-resources to return, these endpoints take ``count`` and either ``min`` or
-``max`` query parameters. They specify the total number of items to include, as
-well as what the first or last item in the list should be. For convenience,
-both *min* and *max* may refer to either an index or a resource id.
-
-*Example paged request URL:*
-
-.. code-block:: text
-
-   http://rest.api.domain/blocks?count=100&min=200
-
-Within the ``"paging"`` property of the response body there will be one or more
-of these four values:
-
-   * **start_index** - index of the first item in the fetched list
-   * **total_count** - total number of resources available
-   * **previous** - URL for the previous page, if any
-   * **next** - URL for the next page, if any
-
-*Example paging response:*
-
-.. code-block:: json
-
-   {
-     "data": [{"fetched": "resources"}],
-     "paging": {
-       "start_index": 200,
-       "total_count": 54321,
-       "previous": "http://rest.api.domain/state?head=65cd...47dd&count=100&min=100",
-       "next": "http://rest.api.domain/state?head=65cd...47dd&count=100&min=300"
-     }
-   }
-
-
-Errors
-------
-
-If something goes wrong while processing a request, the REST API will send back
-a response envelope with only one property: ``"error"``. That error will contain
-three values which explain the problem that occurred:
-
-   * **code** - machine-parsable code specific to this particular error
-   * **title** - short human-readable headline for the error
-   * **message** - longer more detailed explanation of what went wrong
-
-*Example error response:*
-
-.. code-block:: json
-
-   {
-     "error": {
-       "code": 30,
-       "title": "Submitted Batches Invalid",
-       "message": "The submitted BatchList is invalid. It was poorly formed, or has an invalid signature."
-     }
-   }
+==========
 
-.. note::
-
-   While the title or message of an error may change or be reworded over time,
-   **the code is fixed**, and will always refer to the same error.
-
-
-Query Parameters
-================
-
-Many routes support query parameters to help specify how a request to the
-validator should be formed. Not every endpoint supports every query, and some
-endpoints have their own parameters specific to just to them. Any queries
-specific to a single endpoint are not listed here.
-
-.. list-table::
-   :widths: 8, 72
-
-   * - **head**
-     - The id of the block to use as the chain head. This is particularly
-       useful to request older versions of state *(defaults to the latest chain
-       head)*.
-   * - **count**
-     - For paging, this item specifies the number of resources to fetch
-       *(defaults to 1000)*.
```

Comment:
This should be 100

Reply:
This comment is on deleted content that doesn't exist in the current version.

Reply:
I'm bad, this is deleted code.